File: src/main/java/io/github/takenoha/towerdefense/paper/TowerEntityIdentity.java
package io.github.takenoha.towerdefense.paper;

import io.github.takenoha.towerdefense.domain.TowerType;
import java.util.Objects;
import java.util.UUID;

/** PDC identity carried by the persistent Paper entity representing a tower. */
public record TowerEntityIdentity(UUID towerId, UUID teamId, TowerType type, int individualLevel) {
    public TowerEntityIdentity {
        Objects.requireNonNull(towerId, "towerId");
        Objects.requireNonNull(teamId, "teamId");
        Objects.requireNonNull(type, "type");
        if (individualLevel <= 0) {
            throw new IllegalArgumentException("individualLevel must be positive");
        }
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/TowerEntityIdentity.kt
package io.github.takenoha.towerdefense.paper

import io.github.takenoha.towerdefense.domain.TowerType
import java.util.Objects
import java.util.UUID
import kotlin.jvm.JvmRecord

/** PDC identity carried by the persistent Paper entity representing a tower. */
@JvmRecord
data class TowerEntityIdentity(
    val towerId: UUID,
    val teamId: UUID,
    val type: TowerType,
    val individualLevel: Int,
) {
    init {
        Objects.requireNonNull(towerId, "towerId")
        Objects.requireNonNull(teamId, "teamId")
        Objects.requireNonNull(type, "type")
        require(individualLevel > 0) { "individualLevel must be positive" }
    }
}
